app/goal_routes.py

```diff
@@ -0,0 +1,96 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.goal import Goal
+from app import db
+from .task_routes import validate_model
+from app.models.task import Task
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    new_goal = Goal.from_dict(request_body)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response(jsonify(
+        {"goal": {
+            "id" : new_goal.goal_id,
+            "title" : new_goal.title
+        }}
+    ), 201) 
+
+
+@goal_bp.route("", methods=["GET"])
+def get_goals():
+    goals = Goal.query.all()
+    goals_response = [goal.to_dict() for goal in goals]
+
+    return jsonify(goals_response)
+
+
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    return jsonify({"goal": goal.to_dict()}), 200
+
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json() 
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    return jsonify({"goal": goal.to_dict()}), 200
+
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response(jsonify({"details" : f"Goal 1 \"{goal.title}\" successfully deleted"})), 200
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def tasks_to_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    
+    goal.tasks = []
+    for task_id in request_body["task_ids"]:
+        task = validate_model(Task, task_id)
+        goal.tasks.append(task)
+
+    db.session.commit()
+
+    return jsonify({
+        "id" : goal.goal_id,
+        "task_ids" : request_body["task_ids"]
```

Comment:
Instead of using `request_body["task_ids"]` from the client's request to echo back what we received, we could get the task ids from `goal` itself.

You could create a helper method in goal.py to grab the task_ids from an instance of goal:
```python
 def task_ids(self):
        return [task.id for task in self.tasks]
```